src/Bicep.LangServer/Completions/BicepCompletionProvider.cs

```diff
@@ -82,11 +85,19 @@ private IEnumerable<CompletionItem> GetDeclarationCompletions(BicepCompletionCon
 
                 foreach (Snippet resourceSnippet in SnippetsProvider.GetTopLevelNamedDeclarationSnippets())
                 {
+                    Dictionary<string, string> properties = new Dictionary<string, string>()
+                    {
+                        { "name", resourceSnippet.Prefix }
+                    };
+                    TelemetryEvent telemetryEvent = new(TelemetryConstants.EventNames.TopLevelDeclarationSnippetInsertion, properties);
```

Comment:
Would it be worth creating a static method to set the properties, rather than exposing the constructor - that way as we add telemetry events, they're all contained in one place, and enforcing more control over the structure of the dictionary?

Thinking something like:
```csharp
var telemetryEvent = TelemetryEvent.CreateTopLevelDeclarationSnippetInsertion(name: resourceSnippet.Prefix)
```

Reply:
@anthony-c-martin, I refactored this class a bit. Let me know what you think:
https://github.com/Azure/bicep/commit/5078ab95e24bf91942cf55b2072964909d7bfa06